selftests/functional/test_basic.py

```diff
@@ -266,7 +266,7 @@ def test_hanged_test_with_status(self):
                                                "ERROR", res))
             self.assertIn("Test reported status but did not finish",
                           results["tests"][0]["fail_reason"])
-            self.assertLess(res.duration, 40, "Test execution took too long, "
+            self.assertLess(res.duration, 5, "Test execution took too long, "
```

Comment:
This has the strong potential to result in false positives in some environments, as seen in many other time sensitive examples.  Again, we need to follow a trend on those timing related tests.